frontend/src/components/module-editor/content/EditableContentBlock.tsx

```diff
@@ -2,35 +2,52 @@ import { Box, Divider, Flex, useDisclosure, VStack } from "@chakra-ui/react";
 import React, { useState } from "react";
 import { Draggable } from "react-beautiful-dnd";
 
-import { ContentBlock, ContentTypeEnum } from "../../types/ModuleEditorTypes";
-import { TextBlock, ImageBlock } from "../common/content";
-import EditContentOptionsMenu from "./EditContentOptionsMenu";
+import {
+  ContentBlockProps,
+  ContentBlockState,
+} from "../../../types/ContentBlockTypes";
+import { TextBlock, ImageBlock } from "../../common/content";
+import EditContentOptionsMenu from "../EditContentOptionsMenu";
 
-import { ReactComponent as DragHandleIconSvg } from "../../assets/DragHandle.svg";
-import EditTextModal from "./EditTextModal";
+import { ReactComponent as DragHandleIconSvg } from "../../../assets/DragHandle.svg";
+import { EditImageModal, EditTextModal } from "./modals";
+import ContentBlockRenderer from "../../common/content/ContentBlockRenderer";
+import { EditContentModalProps } from "../../../types/ModuleEditorTypes";
 
 /* eslint-disable react/jsx-props-no-spreading */
 
-const SelectContentBlock = (block: ContentBlock): React.ReactElement => {
-  const { type, content } = block;
+const Empty = () => null;
+const CONTENT_BLOCKS = new ContentBlockRenderer<
+  ContentBlockProps<ContentBlockState>
+>({
+  column: Empty,
+  heading: Empty,
+  text: TextBlock,
+  link: Empty,
+  button: Empty,
+  image: ImageBlock,
+  video: Empty,
+  audio: Empty,
+});
 
-  switch (type) {
-    case ContentTypeEnum.TEXT:
-      return <TextBlock content={content} />;
-    case ContentTypeEnum.IMAGE:
-      return <ImageBlock content={content} />;
-    default:
-      throw Error(
-        `Unknown content type given to EditableContentBlock: "${type.title}"`,
-      );
-  }
-};
+const EDIT_MODALS = new ContentBlockRenderer<
+  EditContentModalProps<ContentBlockState>
+>({
+  column: Empty,
+  heading: Empty,
+  text: EditTextModal,
+  link: Empty,
+  button: Empty,
+  image: EditImageModal,
+  video: Empty,
+  audio: Empty,
+});
```

Comment:
It could be nice to define one large table with: 
1. Content type (e.g. text)
2. Content block component (e.g. TextBlock)
3. Edit modal (e.g. EditTextModal)

Then we could automagically split this larger table up into the two ContentBlockRenderer objects you have here. That way people can create a new component and put everything into one "row" of the table as opposed to editing multiple things in the content block. Is it worth the possible additional complexity? Idk, but worth a thought.